static/app/utils/useKeyPress.tsx

```diff
@@ -3,31 +3,39 @@ import {useEffect, useState} from 'react';
 /**
  * Hook to detect when a specific key is being pressed
  */
-function useKeyPress(targetKey: React.KeyboardEvent['key']) {
+const useKeyPress = (
+  targetKey: string,
+  targetRef?: React.RefObject<HTMLInputElement>
+) => {
   const [keyPressed, setKeyPressed] = useState(false);
 
   useEffect(() => {
-    function downHandler({key}: KeyboardEvent) {
+    const downHandler = ({key}: KeyboardEvent) => {
       if (key === targetKey) {
         setKeyPressed(true);
       }
-    }
+    };
 
-    function upHandler({key}: KeyboardEvent) {
+    const upHandler = ({key}: KeyboardEvent) => {
       if (key === targetKey) {
         setKeyPressed(false);
       }
+    };
+
+    let current: any = window;
+    if (targetRef?.current) {
+      current = targetRef?.current;
     }
```

Comment:
Why doesn't `const current targetRef?.current ?? window` work again?

Reply:
Oh - because TS doesn't like the ambiguity of current being either a targetRef.current or window 🤷 
I'm getting
```
Argument of type '({ key }: KeyboardEvent) => void' is not assignable to parameter of type 'EventListenerOrEventListenerObject'.
  Type '({ key }: KeyboardEvent) => void' is not assignable to type 'EventListener'.
    Types of parameters '__0' and 'evt' are incompatible.
      Type 'Event' is missing the following properties from type 'KeyboardEvent': altKey, charCode, code, ctrlKey, and 17 more.ts(2345)
```

Reply:
Ah.

butts. just realized the only reason why it stopped yelling at me was because i gave it a type of `any` 🤦

Reply:
lemme know if you hate it

Reply:
hmm, what is the type of `current` without the any using `const current = targetRef?.current ?? window`

Reply:
it's either `var window: Window & typeof globalThis` or `(property) RefObject<HTMLInputElement>.current: HTMLInputElement | null | undefined`

Reply:
so which type isn't it happy about?

Reply:
Just typed it `as HTMLElement`
Let me know if you're cool with that

Reply:
I wonder if it would be better to just change `window` to `document.body` so it truly is an HTMLElement

Reply:
🤔 

I can do that 👍

Reply:
yeahI think that should be OK

Reply:
niice
So tested and this all works as expected :D 

good catch on the document body 😂